extensions/kafka-extraction-namespace/src/test/java/io/druid/query/extraction/namespace/TestKafkaExtractionCluster.java

```diff
@@ -373,4 +399,55 @@ public void testSimpleRename() throws InterruptedException
       producer.close();
     }
   }
+
+  private void assertUpdated(
+      String expected,
+      String namespace,
+      String key,
+      ConcurrentMap<String, Function<String, String>> lookup
+  )
+      throws InterruptedException
+  {
+    final Function<String, String> extractionFn = lookup.get(namespace);
+
+    int count = 0;
+    if (expected == null) {
+      while (extractionFn.apply(key) != null) {
```

Comment:
Instead of having a timeout here, how about having this never timeout and instead applying a timeout to the test? `@Test(timeout = blah)`. Seems like it would be more robust to have one big timeout than a lot of little timeouts that would be more sensitive to noise.